File: CoreTransformer/StateClass.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

''' <summary>
''' State Class
''' </summary>
''' <remarks></remarks>
Class StateClass

    Private m_Id As Integer
    Private m_StateLabelXID As Integer
    Private m_StateLabelYID As Integer

    Public Sub New(
        ByVal id As Integer,
        ByVal stateLabelXID As Integer,
        ByVal stateLabelYID As Integer)

        Me.m_Id = id

        Me.m_StateLabelXID = stateLabelXID
        Me.m_StateLabelYID = stateLabelYID

    End Sub

    Public ReadOnly Property Id As Integer
        Get
            Return Me.m_Id
        End Get
    End Property

    Public ReadOnly Property StateLabelXID As Integer
        Get
            Return Me.m_StateLabelXID
        End Get
    End Property

    Public ReadOnly Property StateLabelYID As Integer
        Get
            Return Me.m_StateLabelYID
        End Get
    End Property

End Class

File: src/Runtime/StateClass.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2023 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

namespace SyncroSim.STSim
{
    internal class StateClass
    {
        private int m_Id;
        private int m_StateLabelXId;
        private int m_StateLabelYId;
        private string m_DisplayName;

        public StateClass(
            int id, 
            int stateLabelXId, 
            int stateLabelYId, 
            string displayName)
        {
            this.m_Id = id;
            this.m_StateLabelXId = stateLabelXId;
            this.m_StateLabelYId = stateLabelYId;
            this.m_DisplayName = displayName;
        }

        public int Id
        {
            get
            {
                return this.m_Id;
            }
        }

        public int StateLabelXID
        {
            get
            {
                return this.m_StateLabelXId;
            }
        }

        public int StateLabelYID
        {
            get
            {
                return this.m_StateLabelYId;
            }
        }

        public string DisplayName
        {
            get
            {
                return this.m_DisplayName;
            }
        }
    }
}
